src/core/Akka/Actor/Futures.cs

```diff
@@ -129,20 +129,20 @@ public static async Task<T> Ask<T>(this ICanTell self, Func<IActorRef,object> me
         }
 
         /// <summary>
-        /// TBD
+        /// Resolves <see cref="IActorRefProvider"/> for Ask pattern
         /// </summary>
-        /// <param name="self">TBD</param>
-        /// <returns>TBD</returns>
+        /// <param name="self">Reference to someone we are sending Ask request to</param>
+        /// <returns>Provider used for Ask pattern implementation</returns>
         internal static IActorRefProvider ResolveProvider(ICanTell self)
         {
-            if (ActorCell.Current != null)
-                return InternalCurrentActorCellKeeper.Current.SystemImpl.Provider;
+            if (self is ActorSelection)
+                return ResolveProvider(self.AsInstanceOf<ActorSelection>().Anchor);
 
             if (self is IInternalActorRef)
                 return self.AsInstanceOf<IInternalActorRef>().Provider;
-
-            if (self is ActorSelection)
-                return ResolveProvider(self.AsInstanceOf<ActorSelection>().Anchor);
+            
+            if (ActorCell.Current != null)
+                return InternalCurrentActorCellKeeper.Current.SystemImpl.Provider;
```

Comment:
So the reason of the issue was the following:

Usually, using `ActorCell.Current` works when since it is some actor from Sys2 requesting `Ask` to actor of Sys1. But here, `ActorCell.Current` is `akka://test/temp/XXX`, the test actor of test system. This test actor's path is not resolvable for remote actors of Sys1 or Sys2 (test `Sys` is not configured to use remoting and does not have association with any of this systems).

Checks ordering change here allows to use correct provider (RemoteRefProvider in this case) with `Sys2` system for hosting temporary actor.

Reply:
WOW. Nice catch. I think this error might also have a lot to do with some of the "unable to resolve X" errors that occur in remote Ask operations too. I'm going to do some more testing around this locally, but this is an amazing find.

Reply:
Yeah, that could lead to such "unable to resolve" errors I think.